src/contribution-collector.js

```diff
@@ -206,17 +208,18 @@ export async function getContributorsList() {
 							"dotOrg"
 						)
 					) {
-						contributors.unlinked.add(username);
+						contributorLists['unlinked'].push(username);
 						return;
 					}
 
-					return `Co-Authored-By: ${username} <${dotOrg}@git.wordpress.org>`;
+					return contributorLists['coAuthored'].push( `Co-Authored-By: ${username} <${dotOrg}@git.wordpress.org>` );
```

Comment:
We can, but because our ultimate goal is to give credit in the Credits API, I chose to use the `user@git.wordpress.org` format. I also didn't love the idea of hard coding personal emails in the commit messages if someone has that setting turned off. I don't recall if the anonymous format worked even if you had that disabled. I could test if there's a strong preference to use that format instead.

The other thought I had was for consistency with how committers are attributed credit from the WordPress/wordpress-develop mirror. When a contributor is granted commit access, they are told to add the `@git.wordpress.org` format to their account, as that's the email given to the commit author. A "pie in the sky" idea that I'll investigate once we get this initial round settled is to include `Co-Authored-By` trailers for everyone that receives Props in an SVN commit. My thinking there is that the same format as the committer would make sense.

By including unlinked contributors in the commit message, though, we can easily extract their GH slug from the list and look in the .org database for any contributors that have connected their account since any of their contributions were merged.

There's some weirdness to detail around those emails as noted on that page:
> If you created your account on GitHub.com after July 18, 2017, your noreply email address for GitHub is an ID number and your username in the form of [ID+USERNAME@users.noreply.github.com](mailto:ID+USERNAME@users.noreply.github.com). If you created your account on GitHub.com prior to July 18, 2017, and enabled Keep my email address private prior to that date, your noreply email address from GitHub is [USERNAME@users.noreply.github.com](mailto:USERNAME@users.noreply.github.com).
In my reasearch from a year or so ago, I couldn't find a way to figure out which format a specific user has. But in recent testing, I found that both formats worked, regardless of the format it tells you you have.

Another layer to that is on the [Change your username](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/changing-your-github-username#your-git-commits) page:

> If you've been using a GitHub-provided private commit email address, whether or not your commit history will be retained after an account rename depends on the format of the email address. Git commits that are associated with your GitHub-provided noreply email address won't be attributed to your new username and won't appear in your contributions graph, unless your noreply email address is in the form of ID+USERNAME@users.noreply.github.com. Older versions of the noreply email address that do not contain a numeric ID will not be associated with your GitHub account after changing your username.

Sorry for the brain dump, but this was a good chance to document some of my thinking/findings.

Reply:
Given the goal here is to improve the breadth of folks properly credited from GitHub repos back to WordPress core major releases and to streamline the major release credits gathering process, the approach that @desrosj has here makes sense.  If the goal was to ensure all GitHub users were properly credited on PRs, then using the format @peterwilsoncc noted would likely be the best approach.

Reply:
Been thinking about this the last few days.

I think I do prefer using all .org usernames, but it could be reasonable to use the GH format email for any contributor that's not linked to a .org one.

We can still parse out their GH user and look them up in the .org API, and they'd receive attribution on GH from their anonymous email. But that also removes some of the incentive to connect their account in the first place.

Reply:
@desrosj Does the WP.org integration add `username@git.wordpress.org` to a users profile when they link their accounts or is that a step the contributor would need to do to get recognition on GitHub?

Reply:
Currently, they will need to add it manually. I plan to include this in the announcement post, and the documentation. Based on [this comment ](https://make.wordpress.org/core/2021/05/03/expired-github-and-wordpress-org-profile-connections/#comment-41291), though, it doesn't seem that our authentication request currently allows us to modify your profile at all. We'd need to add that to the request, and contributors with pre-existing connections would need to re-approve the connection with the new permissions.

I was talking this through with @dd32 last night. Wanted to document our discussion. At first he did not like using `@git.wp.org` emails, but after some discussion, this was how he summarized his thoughts:

- > On a GitHub-first repo, it makes sense to use @…[github.com](http://github.com/) emails
- > On a SVN repo, it makes sense to use username
- > On a SVN -> Git sync, it makes sense to append Co-Authored-by: …@git.wordpress.org  (But who knows if we wanna do that)
- > On a SVN -> Git -> GitHub sync, We can’t modify the commit message of ^ so meh, i guess it makes sense to use @git.w.org
- > Because of ^, it makes sense that on a GitHub-First repo, we could just use @git.w.org for consistency.

It's worth noting that in that list, there are currently no SVN -> Git syncs that do _not_ make it over to GitHub.

Some other takeaways from our discussion:
- Even though we're heavily using GitHub at this point, there's always a non-0 chance that as a project we decide to use something other than GitHub (ie. GitLab). If this does happen, then the `ID+username@github` emails would be out of place. We could likely add a way for the tool to support these, but using wp.org ones would be somewhat tool-agnostic.
- There's currently no ability to validate that you own the `@git.wp.org` email that represents you. Adding the ability to verify this through email will likely be appropriate to avoid any problems.

Reply:
I've gone and created #28 for changing the unlinked contributor list to use GH emails. I think that makes sense, but figured it would be good to tackle separately.